lib/core/scheduler.rb

```diff
@@ -119,7 +112,12 @@ def sync_triggered?(connector_settings)
         return false
       end
 
-      next_trigger_time = cron_parser.next_time(Time.parse(last_synced))
+      last_synced = begin
+        Time.parse(connector_settings[:last_synced])
+      rescue StandardError
```

Comment:
Do I get it right that you handling a case when `connector_settings[:last_synced]` is empty?

If yes, then I have a question. Do you also want to replace the current `connector_settings[:last_synced]` value by `Time.now` if it's stored in the incorrect format? e.g. 

```Ruby
irb(main):004:0> Time.parse('foo')
Traceback (most recent call last):
        7: from /Users/vidok/.rbenv/versions/2.6.9/bin/irb:23:in `<main>'
        6: from /Users/vidok/.rbenv/versions/2.6.9/bin/irb:23:in `load'
        5: from /Users/vidok/.rbenv/versions/2.6.9/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        4: from (irb):4
        3: from (irb):4:in `rescue in irb_binding'
        2: from /Users/vidok/.rbenv/versions/2.6.9/lib/ruby/2.6.0/time.rb:375:in `parse'
        1: from /Users/vidok/.rbenv/versions/2.6.9/lib/ruby/2.6.0/time.rb:194:in `make_time'
ArgumentError (no time information in "foo")
```

Reply:
I didn't think of this scenario, but I think it's a good practice.

`last_synced` is set by the connector service, not operators, so I don't expect any surprise. 

Of course, this can be tampered via Dev Tools or ES API. If that happens, I think it also makes sense to use `Time.now` as a fallback. cc @danajuratoni